sound/soc/sof/intel/telemetry.c

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2023 Intel Corporation. All rights reserved.
+//
+//
+
+#include <sound/sof/ipc4/header.h>
+#include <sound/sof/xtensa.h>
+#include "../ipc4-priv.h"
+#include "../sof-priv.h"
+#include "hda.h"
+#include "telemetry.h"
+
+#define CHECK_SEPARATOR(S, V) \
+	if (S != V) { \
+		dev_err(sdev->dev, "error: %s : %#x is not matched with %#x", #S, S, V); \
+		return false; \
+	} \
```

Comment:
this should be wrapped in `do {} while (0)` to make sure the semicolon after this macro belongs to it and not treated as a separate statement.

Reply:
good idea! thanks